packages/thirdweb/src/react/web/ui/ConnectWallet/screens/ViewNFTs.tsx

```diff
@@ -0,0 +1,241 @@
+import { useQueries } from "@tanstack/react-query";
+import { useMemo } from "react";
+import type { Chain } from "../../../../../chains/types.js";
+import { getCachedChain } from "../../../../../chains/utils.js";
+import type { ThirdwebClient } from "../../../../../client/client.js";
+import { getContract } from "../../../../../contract/contract.js";
+import { getOwnedNFTs as getErc721OwnedNFTs } from "../../../../../extensions/erc721/read/getOwnedNFTs.js";
+import { isERC721 } from "../../../../../extensions/erc721/read/isERC721.js";
+import { getOwnedNFTs as getErc1155OwnedNFTs } from "../../../../../extensions/erc1155/read/getOwnedNFTs.js";
+import { isERC1155 } from "../../../../../extensions/erc1155/read/isERC1155.js";
+import type { SupportedNFTs } from "../../../../core/utils/defaultTokens.js";
+import { useActiveAccount } from "../../../hooks/wallets/useActiveAccount.js";
+import { useActiveWalletChain } from "../../../hooks/wallets/useActiveWalletChain.js";
+import { MediaRenderer } from "../../MediaRenderer/MediaRenderer.js";
+import { Skeleton } from "../../components/Skeleton.js";
+import { Spacer } from "../../components/Spacer.js";
+import { Container, Line, ModalHeader } from "../../components/basic.js";
+
+const fetchNFTs = async (
+  client: ThirdwebClient,
+  chain: Chain,
+  nftAddress: string,
+  owner: string,
+) => {
+  const contract = getContract({
+    address: nftAddress,
+    chain,
+    client,
+  });
+
+  const erc721 = await isERC721({ contract }).catch(() => {
+    throw new Error(
+      `Failed to read contract bytecode for NFT ${nftAddress} on ${chain.name || chain.id}, is this NFT on the correct chain?`,
+    );
+  });
+  if (erc721) {
+    const result = await getErc721OwnedNFTs({
```

Comment:
oof i just remembered this is not implemented in our regular drop contracts iirc - will prob fail for most drops. we should try it

Reply:
It requires either tokenOfOwnerByIndex (which our default collection has) or getOwnedTokenIds (which the queryable extension has). The latter I added to `getNFTs` in the same PR. If the contract doesn't support either it logs an error and hides the nft